lib/srv/db/common/auth.go

```diff
@@ -70,7 +70,7 @@ type Auth interface {
 // AuthConfig is the database access authenticator configuration.
 type AuthConfig struct {
 	// AuthClient is the cluster auth client.
-	AuthClient *libauth.Client
+	AuthClient libauth.ClientI
```

Comment:
I'm not sure I like this change, I'd rather remove ClientI than spread it.

Ditto for similar changes.

Reply:
Agreed, using `*auth.Client` makes some testing difficult though. However I don't need this anymore since I updated the tests.